source/common/http/header_utility.cc

```diff
@@ -235,7 +235,96 @@ bool HeaderUtility::headerNameContainsUnderscore(const absl::string_view header_
   return header_name.find('_') != absl::string_view::npos;
 }
 
+namespace {
+// This function is used to validate the authority.
+// Because it is used for both H/1 and H/2 validations, this function permits
```

Comment:
Optional: I think expanding out  HTTP/1 and HTTP/2 in the comment text would be helpful for future readers.

Reply:
I tried to rephrase the comment, and make it more explicit. Let me know if further clarification is desired.